### DISCUSSION TITLE
Refactoring for creating dummy instances for catching parsing errors in Match Lines

### DISCUSSION THREAD

**Discussion by CarlosCancino-Chacon** [CPJKU/partitura#333] [Ideas]
In PR #332, dummy instances of match lines are created inside the `prepare_kwargs_from_matchline` method, in order to fix issues with regular expression pattern checking (and raising `MatchError`s).  It would be more efficient to create the dummy instances of the match lines outside of the method (i.e., as global variables), otherwise, the new instances of  `dummy_snote`and  `dummy_instance` will slow down the parsing of match files.

However, as pointed by @sildater, creating dummy instances for checking match errors would require a larger refactoring and potentially introduce other issues. The current solution in PR #332 will stay for the time being, but if this solution ends up slowing down pipelines that require the loading of many match files (e.g., from the (n)ASAP dataset), we should consider a refactoring.

**Comment by sildater** [comment]
maybe PR #334 addresses this with minimal changes
